ksp_plugin/plugin.cpp

```diff
@@ -769,13 +769,14 @@ void Plugin::WaitForVesselToCatchUp(PileUpFuture& pile_up_future,
   PileUp const* const pile_up = pile_up_future.pile_up;
   auto& future = pile_up_future.future;
   future.wait();
-  if (future.get().error() == Error::OUT_OF_RANGE) {
+  if (!future.get().ok()) {
     for (not_null<Part*> const part : pile_up->parts()) {
       not_null<Vessel*> const vessel =
           FindOrDie(part_id_to_vessel_, part->part_id());
       if (collided_vessels.insert(vessel).second) {
-        LOG(INFO) << "Vessel " << vessel->ShortDebugString()
-                  << " collided with a celestial";
+        LOG(WARNING) << "Vessel " << vessel->ShortDebugString()
+                     << " collided with a celestial: "
+                     << future.get().ToString();
```

Comment:
We cannot call `future.get()` twice, see https://en.cppreference.com/w/cpp/thread/future/get:
> The behavior is undefined if `valid()` is `false` before the call to this function.
Any shared state is released. `valid()` is `false` after a call to this method.